gcc/config/riscv/constraints.md

```diff
@@ -35,6 +35,14 @@
 
 ;; General constraints
 
+(define_memory_constraint "m"
+  "An address that is not base reg + index reg or post modify."
+  (and (match_code "mem")
+       (match_test "!(GET_CODE (XEXP (op, 0)) == PLUS
+                        && GET_CODE (XEXP (XEXP (op, 0), 0)) == REG
+                        && GET_CODE (XEXP (XEXP (op, 0), 1)) == REG)
+                    && GET_CODE (XEXP (op, 0)) != POST_MODIFY")))
```

Comment:
The default `m` constraint does more than just check if code is `mem` https://github.com/openhwgroup/corev-gcc/blob/4a6bad0f495f038f948a00a40464e3d28761b0e4/gcc/common.md#L24-L27

You can also reuse the new CVmp constraint.

Check if this works and is sufficient:
```
(and (match_operand 0 "TARGET_MEM_CONSTRAINT")
      not (match_operand 0 "CVmp"))
```
Hopefully `"TARGET_MEM_CONSTRAINT"` works, I've based it on how it's used in `common.md`.

Reply:
Actually, could you reuse the cv mem predicates e.g.
```
not (match_operand 0 "mem_post_inc"))
```
and
```
not (match_operand 0 "mem_plus_reg"))
```